aws/data_source_aws_workspaces_image.go

```diff
@@ -0,0 +1,64 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsWorkspacesImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsWorkspacesImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"image_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"required_tenancy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsWorkspacesImageRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).workspacesconn
+
+	imageID := d.Get("image_id").(string)
+	input := &workspaces.DescribeWorkspaceImagesInput{
+		ImageIds: []*string{aws.String(imageID)},
+	}
+
+	resp, err := conn.DescribeWorkspaceImages(input)
+	if err != nil {
+		return err
+	}
```

Comment:
Will probably want to check `len(resp.Images) > 0` here.
See https://github.com/terraform-providers/terraform-provider-aws/pull/11837.

Reply:
Good point, will add